### PR TITLE
Added loading indicator when approving ledger public key

### PR DESCRIPTION
**Pull request by amirsaran3**

### CODE REVIEW

File: src/modal/Modal.tsx [hunk 1]
```diff
@@ -167,8 +167,12 @@ const Modal: React.FC = () => {
...
                 Dismiss
               </button>
-              <button className="right-button" onClick={handleConnectClick}>
-                Connect
+              <button
+                className="right-button"
+                onClick={handleConnectClick}
+                disabled={state.loading}
+              >
+                {state.loading ? "Loading..." : "Connect"}
```
Comment (lewis-sqa) [inline src/modal/Modal.tsx:175]: We should just use `useState` inside the component for whether we're loading. It's purely UI state. We should also disable the dismiss button and make the inputs read-only

**Review by lewis-sqa — CHANGES_REQUESTED** [review]

File: src/modal/Modal.tsx (hunk 1 above)
Comment (amirsaran3) [reply src/modal/Modal.tsx]: Resolved issues and committed

File: src/modal/Modal.tsx [hunk 2]
```diff
@@ -75,12 +76,16 @@ const Modal: React.FC = () => {
...
+    setIsLoading(true);
     const wallet = state.walletProviders["ledgerwallet"] as ILedgerWallet;
 
     wallet.setDerivationPath(ledgerDerivationPath);
     wallet.setAccountId(ledgerAccountId);
 
-    wallet.signIn().catch((err) => setLedgerError(`Error: ${err.message}`));
+    await wallet
+      .signIn()
+      .catch((err) => setLedgerError(`Error: ${err.message}`));
```
Comment (lewis-sqa) [inline src/modal/Modal.tsx:87]: We'll want to set loading to `false` if there's an error too

File: src/modal/Modal.tsx [hunk 3]
```diff
@@ -160,15 +166,24 @@ const Modal: React.FC = () => {
...
                 Dismiss
               </button>
-              <button className="right-button" onClick={handleConnectClick}>
-                Connect
+              <button
+                className="right-button"
+                onClick={handleConnectClick}
+                disabled={isLoading}
+              >
+                {isLoading ? "Loading..." : "Connect"}
```
Comment (lewis-sqa) [inline src/modal/Modal.tsx:186]: I know I previously suggested "Loading..." but maybe "Connecting..." would be better here until we have a spinner. Not sure if it causes any reflow problems with it being a longer string?

File: src/modal/Modal.tsx (hunk 2 above)
Comment (lewis-sqa) [reply src/modal/Modal.tsx:87]: Ah... We get around this because the `catch` suppresses the Promise rejection

**Review by lewis-sqa — APPROVED** [review]

**Review by lewis-sqa — CHANGES_REQUESTED** [review]
Sorry, I did some testing on this branch. We've missed disabling the close button and the backdrop. It causes a situation where I can close the modal and later trigger `signIn` with my Ledger that's waiting for me to approve the public key transaction.

**Review by lewis-sqa — APPROVED** [review]

**Pull request merged**